scene/resources/curve.cpp

```diff
@@ -1650,33 +1662,20 @@ void Curve3D::_bake() const {
 	}
 
 	// Step 2: Calculate the up vectors and the whole local reference frame
-	//
-	// See Dougan, Carl. "The parallel transport frame." Game Programming Gems 2 (2001): 215-219.
-	// for an example discussing about why not the Frenet frame.
-	{
-		int point_count = baked_point_cache.size();
+	int point_count = baked_point_cache.size();
+	baked_up_vector_cache.resize(point_count);
+	Vector3 *up_write = baked_up_vector_cache.ptrw();
+	const Vector3 *forward_ptr = baked_forward_vector_cache.ptr();
+	const Vector3 *points_ptr = baked_point_cache.ptr();
 
-		baked_up_vector_cache.resize(point_count);
-		Vector3 *up_write = baked_up_vector_cache.ptrw();
-
-		const Vector3 *forward_ptr = baked_forward_vector_cache.ptr();
-		const Vector3 *points_ptr = baked_point_cache.ptr();
+	if (bake_mode == BAKE_PARALLEL_TRANSPORT) {
+		// See Dougan, Carl. "The parallel transport frame." Game Programming Gems 2 (2001): 215-219. for a discussing about its difference with Frenet frame.
 
 		Basis frame; // X-right, Y-up, Z-forward.
 		Basis frame_prev;
 
 		// Set the initial frame based on Y-up rule.
-		{
-			Vector3 forward = forward_ptr[0];
-
-			if (abs(forward.dot(Vector3(0, 1, 0))) > 1.0 - UNIT_EPSILON) {
-				frame_prev = Basis::looking_at(forward, Vector3(1, 0, 0));
-			} else {
-				frame_prev = Basis::looking_at(forward, Vector3(0, 1, 0));
-			}
-
-			up_write[0] = frame_prev.get_column(1);
-		}
+		up_write[0] = _ensure_valid_up_vector(forward_ptr[0], Vector3(0, 1, 0), Vector3(1, 0, 0));
```

Comment:
You're not updating `frame_prev` here, this breaks the rest of the code below

Reply:
right, that's why the tests break. fixed